### PR TITLE
[REVIEW] column reduction benchmark

### PR DESCRIPTION
**Pull request by karthikeyann**
closes https://github.com/rapidsai/cudf/issues/5226

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: cpp/benchmarks/reduction/reduce_benchmark.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+#include <memory>
+#include <random>
+
+class Reduction : public cudf::benchmark {
+};
+
+// TODO: put it in a struct so `uniform` can be remade with different min, max
+template <typename T>
+T random_int(T min, T max)
```
Comment (devavret) [inline cpp/benchmarks/reduction/reduce_benchmark.cpp:33]: This was the old style. There’s a better way now: https://github.com/rapidsai/cudf/blob/ab3911d880702ed085d6d45dcadd32dcec02e068/cpp/benchmarks/search/search_benchmark.cu#L116

File: cpp/benchmarks/reduction/reduce_benchmark.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  REDUCE_BENCHMARK_DEFINE(double, aggregation);
+
+REDUCE_BENCHMARK_NUMERIC(sum);
+REDUCE_BENCHMARK_NUMERIC(product);
+REDUCE_BENCHMARK_NUMERIC(min);
+using cudf::timestamp_ms;
+REDUCE_BENCHMARK_DEFINE(timestamp_ms, min);
+REDUCE_BENCHMARK_NUMERIC(mean);
+REDUCE_BENCHMARK_NUMERIC(variance);
+REDUCE_BENCHMARK_NUMERIC(std);
```
Comment (harrism) [inline cpp/benchmarks/reduction/reduce_benchmark.cpp:101]:
```suggestion
REDUCE_BENCHMARK_NUMERIC(std);

```

**Comment by harrism** [comment]
CI failure is because the code is still using the `experimental` namespace.

```
/var/lib/jenkins/workspace/rapidsai/gpuci/cudf/prb/cudf-gpu-build_4/cpp/benchmarks/reduction/reduce_benchmark.cpp:34:41: error: 'experimental' is not a member of 'cudf'

                   std::unique_ptr<cudf::experimental::aggregation> const& agg)
```

**Pull request merged**